server/@types/hmpo-form-wizard/index.d.ts

```diff
@@ -1,7 +1,8 @@
 // Copied from https://github.com/ministryofjustice/hmpps-strengths-based-needs-assessments-ui/blob/c1d2e0ac64ce2be0ea0a27919cd34f8882b2d5c0/server/%40types/hmpo-form-wizard/index.d.ts
```

Comment:
could you replace this comment with the reference to wherever the type updates come from pls?

the latest ones in `hmpps-strengths-based-needs-assessments-ui` are quite different.

Reply:
_**FWIW**_ I've noticed some types weirdness related to `Step` for example. The `hmpo-form-wizard` documentation and examples suggest pretty much all fields are optional?

[from the docs: ](https://github.com/HMPO/hmpo-form-wizard/tree/master?tab=readme-ov-file#additional-step-options)
> The minimum amount of configuration for a wizard step is the next property to determine where the user should be taken after completing a step. A number of additional properties can be defined.
> [...]

Also some of [their examples](https://github.com/HMPO/hmpo-form-wizard/tree/master?tab=readme-ov-file#usage) have steps with nothing in them, e.g.

```
// steps.js
module.exports = {
  '/step1': {
    next: 'step2'
  },
  // [...]
  '/step4': {},
  '/not-old-enough': {}
}
```

🤷

Reply:
Updating the link now. 
In terms of the types, yeah it is a bit strange in that aspect as I believe it is very flexible as most if not all fields are not necessarily needed to actually generate and show something, but then lots can be added for additional configurability.